lab-david/model/user.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const crypto = require('crypto');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const mongoose = require('mongoose');
+const createErrors = require('http-errors');
+const Promise = require('bluebird');
+const debug = require('debug')('cfgram:user');
+
+const userSchema = mongoose.Schema({
+  username: {type: String, required: true, unique: true},
+  email: {type: String, required: true, unique: true},
+  password: {type: String, required: true},
+  findHash: {type: String, unique: true},
+});
+
+userSchema.methods.generatePasswordHash = function(password){
+  debug('generatePasswordHash');
+
+  return new Promise((resolve, reject) => {
+    bcrypt.hash(password, 10, (err, hash) => {
+      if(err) return reject(err);
+      this.password = hash;
+      resolve(this);
+    });
+  });
+};
+
+userSchema.methods.comparePasswordHash = function(password){
+  debug('comparePasswordHash');
+
+  return new Promise((resolve, reject) => {
+    bcrypt.compare(password, this.password, (err, valid) => {
+      if(err){
```

Comment:
shouldn't have to separate if blocks checking the same condition. Also, really don't need these logs here. They just blast the console and make your tests look messy.